src/Lykke.HftApi.Services/Idempotency/IdempotencyService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Threading.Tasks;
+using AzureStorage;
+
+namespace Lykke.HftApi.Services.Idempotency
+{
+    public class IdempotencyService
+    {
+        private readonly INoSQLTableStorage<IdempotentEntity> _tableStorage;
+
+        public IdempotencyService(INoSQLTableStorage<IdempotentEntity> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        }
+
+        public async Task<string> CreateEntityOrGetPayload(string requestId)
+        {
+            var entity = IdempotentEntity.Build(requestId);
+            
+            var createdNow = await _tableStorage.CreateIfNotExistsAsync(entity);
+
+            if (createdNow)
+            {
+                return null;
+            }
+            else
+            {
+                entity = await _tableStorage.GetTopRecordAsync(entity.PartitionKey);
+
+                return entity.Payload;
+            }
+        }
+
+        public async Task UpdatePayload(string requestId, string payload)
+        {
+            var entity = IdempotentEntity.Build(requestId, payload);
+
+            await _tableStorage.InsertOrMergeAsync(entity);
```

Comment:
it should be just inserted, not merged or replaced. If the record already exists an exception should be thrown. A specific error should be returned to the user. The error should indicate that the request should be retried. Something like `CONCURRENT_OPERATION_ERROR` or `OPTIMISITC_CONCURRENCY_ERROR`